CIP-0030/README.md

```diff
@@ -197,6 +197,23 @@ Errors: `APIError`, `PaginateError`
 
 If `amount` is `undefined`, this shall return a list of all UTXOs (unspent transaction outputs) controlled by the wallet. If `amount` is not `undefined`, this request shall be limited to just the UTXOs that are required to reach the combined ADA/multiasset value target specified in `amount`, and if this cannot be attained, `undefined` shall be returned. The results can be further paginated by `paginate` if it is not `undefined`.
 
+### api.getCollateral(params: { amount: cbor\<Coin> }): Promise\<TransactionUnspentOutput[] | null>
+
+Errors: `APIError`
+
+The function takes a required object with parameters. With a single **required** parameter for now: `amount`. (**NOTE:** some wallets may be ignoring the amount parameter, in which case it might be possible to call the function without it, but this behavior is not recommended!). Reasons why the `amount` parameter is required:
+1. Dapps must be motivated to understand what they are doing with the collateral, in case they decide to handle it manually.
+2. Depending on the specific wallet implementation, requesting more collateral than necessarily might worsen the user experience with that dapp, requiring the wallet to make explicit wallet reorganisation when it is not necessary and can be avoided.
+3. If dapps don't understand how much collateral they actually need to make their transactions work - they are placing more user funds than necessary in risk.
+
+So requiring the `amount` parameter would be a by-spec behavior for a wallet. Not requiring it is possible, but not specified, so dapps should not rely on that and the behavior is not recommended.
+
+This shall return a list of all UTXOs (unspent transaction outputs) controlled by the wallet that are required to reach the combined ADA value target specified in `amount` **AND** the best suitable to be used as collateral inputs for transactions with plutus script inputs. If this cannot be attained, `null` shall be returned.
```

Comment:
I would prefer if we returned some kind of "Requirement not met" error so that we can specify which requirement wasn't met (which scales better if we need to add more requirements in the future)

Reply:
Updated the description to `If this cannot be attained, an error message with an explanation of the blocking problem shall be returned.`